client/beefy/src/worker.rs

```diff
@@ -241,7 +247,16 @@ where
 
 				debug!(target: "beefy", "🥩 New Rounds for id: {:?}", id);
 
-				self.best_beefy_block = Some(*notification.header.number());
+				let block_num = *notification.header.number();
+				self.best_beefy_block = Some(block_num);
+				if let Err(err) = self.client.header(BlockId::number(block_num)).map(|h| {
+					if let Some(header) = h {
+						let _ = self.beefy_best_block_sender.notify(|| Ok(header));
+					}
+				}) {
+					error!(target: "beefy", "🥩 Failed to get header for block number {}; err: {:?}",
+						block_num, err);
+				}
```

Comment:
Can't we reuse the header from the notification instead of fetching it from the database again?

Reply:
good catch! now using hash from notification